src/azul/service/index_query_service.py

```diff
@@ -156,12 +157,12 @@ def get_summary(self, catalog: CatalogName, filters):
                 'speciesCount'
             ],
             'projects': [
-                'projectCount',
-                'labCount',
+                'project',
```

Comment:
Be sure you understand the consequences of this. The `project` facet is mapped to the document path `contents.projects.project_short_name`. When a filter for this facet is active, that filter is removed from the top-level filter aggregation for this facet and the `doc_count` property in the corresponding aggregation in the response reflects the count of all documents, not just the ones matching the filter. I don't think that's what we want.

Reply:
I admit I'm having a hard time following this explanation, but more importantly, I can't replicate what I think it's claiming. I have 9 projects indexed on my personal deployment, which is up to date with this PR:

```
$ http https://service.nadove.dev.singlecell.gi.ucsc.edu/version | jq .git.commit
"4fdfbc9f7ce01b2580d06eda375658cfcd8257a6"
$ http 'https://service.nadove.dev.singlecell.gi.ucsc.edu/index/projects' | jq .pagination.total
9
"4fdfbc9f7ce01b2580d06eda375658cfcd8257a6"
$ http 'https://service.nadove.dev.singlecell.gi.ucsc.edu/index/summary' | jq .projectCount
9
```
Filtering by the `project` facet gives the correct result, 1 project:
```
$ http 'https://service.nadove.dev.singlecell.gi.ucsc.edu/index/summary?filters={"project":{"is":["1M Neurons"]}}' | jq .projectCount
1
```
And other filters appear to work as well:
```
$ http 'https://service.nadove.dev.singlecell.gi.ucsc.edu/index/bundles?filters={"project":{"is":["1M Neurons"]}}' | jq .hits[].bundles[].bundleUuid
"444a5c9e-33d4-4adb-b3f3-82f1d8b07f19"
"4440c1ab-978f-4f6c-95f6-3fe54df5eb8b"
 $ http 'https://service.nadove.dev.singlecell.gi.ucsc.edu/index/summary?filters={"bundleUuid":{"is":["444a5c9e-33d4-4adb-b3f3-82f1d8b07f19", "4440c1ab-978f-4f6c-95f6-3fe54df5eb8b"]}}' | jq .projectCount
1
```
And in conjuction:
```
$ http 'https://service.nadove.dev.singlecell.gi.ucsc.edu/index/summary?filters={"bundleUuid":{"is":["444a5c9e-33d4-4adb-b3f3-82f1d8b07f19", "4440c1ab-978f-4f6c-95f6-3fe54df5eb8b"]},"project":{"is":["1M Neurons"]}}' | jq .projectCount
1
```

so if you're correct about the current implementation not doing what we want, I need some help figuring out how to observe that.

Reply:
I agree, it does behave correctly, probably due to the fact that summary uses `post_filter=False`. That's what I was trying to hint at during PL. Each term facet produces a tree of aggregations in the ES request. The root of each tree is a `filter` aggregation populated with the conditions from the `filters` request parameter specified by the client. The conditions used in the `filter` aggregation for facet X exclude any `filters` condition on X itself. Fortunately, the dropped condition has not effect because `post_filter` is False and the results are already filtered via a `query` element in the ES request.

Study `_create_aggregate` and its usages to confirm/deny what I claim. We need someone else other than me to understand how this all works.

If I am right, there is no point in having a filter `aggregation` at the root of each tree when `post_filter` is `False`.

Please confirm that via

```diff
Index: src/azul/service/elasticsearch_service.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/service/elasticsearch_service.py b/src/azul/service/elasticsearch_service.py
--- a/src/azul/service/elasticsearch_service.py	(revision 9cc378e0e840f094d33ccf840c887f60675618cd)
+++ b/src/azul/service/elasticsearch_service.py	(date 1631556653987)
@@ -316,7 +316,7 @@
 
         if enable_aggregation:
             for agg, translation in facet_config.items():
-                es_search.aggs.bucket(agg, self._create_aggregate(catalog, filters, facet_config, agg))
+                es_search.aggs.bucket(agg, self._create_aggregate(catalog, filters if post_filter else {}, facet_config, agg))
 
         return es_search
 
```

and create a FIXME.